src/hosted-buttons/index.js

```diff
@@ -34,24 +39,59 @@ export const getHostedButtonsComponent = (): HostedButtonsComponent => {
         selector,
       });
 
-      // $FlowFixMe
-      Buttons({
-        hostedButtonId,
-        style,
-        onInit,
-        onClick,
-        createOrder: buildHostedButtonCreateOrder({
-          enableDPoP,
-          hostedButtonId,
-          merchantId,
-        }),
-        onApprove: buildHostedButtonOnApprove({
-          enableDPoP,
+      if (shouldRenderSDKButtons(fundingSources)) {
+        const { flexDirection } = getFlexDirection({ ...style });
+
+        buildButtonContainer({ flexDirection, selector });
+
+        // Only render 2 buttons max
+        fundingSources?.slice(0, 2).forEach((fundingSource, index) => {
+          // $FlowFixMe
+          Buttons({
+            hostedButtonId,
+            fundingSource,
+            style: {
+              ...style,
+              color: getButtonColor(style.color, fundingSource),
+            },
+            onInit,
+            onClick,
+            createOrder: buildHostedButtonCreateOrder({
+              enableDPoP,
+              hostedButtonId,
+              merchantId,
+            }),
```

Comment:
option to move these functions above, like:

```js
const createOrder = buildHostedButtonCreateOrder({
    enableDPoP,
    hostedButtonId,
    merchantId,
});
```

and then here it can be:

```js
Buttons({
    ...,
    onInit,
    onClick,
    createOrder,
    onApprove,
})
```

Reply:
edit: this could go one level higher too, like:

```js
const buttonOptions = {
  onInit,
  onClick,
  createOrder,
  onApprove
}
```

and then

```js
Buttons(buttonOptions)
```

totally optional though!